### ISSUE TITLE
cmd/abigen: Vyper issue

### ISSUE THREAD

**Issue by lmittmann** [ethereum/go-ethereum#20418]
I tried to generate ABI bindings for a vyper contract using `abigen` an ran into an issue:

```
$ abigen -v
abigen version 1.9.8-stable
$ vyper --version
0.1.0b14+commit.49b201c
$ abigen --pkg main --type Uniswap --out uniswap.go --vy uniswap_exchange.vy
Fatal: Failed to generate ABI binding: 36:23: expected ';', found '.' (and 10 more errors)
[...forward to line 36...]
	// UniswapExchange.vyABI is the input ABI used to generate the binding from.
	const UniswapExchange.vyABI = "[{...
```

It seems like `abigen` is trying to use the full file name for variable and struct names, including the `.vy` suffix which of course is not allowed in Go. When renaming the file and striping the `.vy` suffix `abigen` works fine.

**Comment by gballet** [comment]
Would you mind sharing your version of `abigen` and `vyper`? Typing the commands you provided doesn't work:

```
Ξ ./abigen --pkg main --type Uniswap --out uniswap.go --vy uniswap_exchange.vy
Contract ABI (--abi), bytecode (--bin) and type (--type) flags are mutually exclusive with the Solidity (--sol) and Vyper (--vy) flags
```

and

```
Ξ ./abigen --pkg main --out uniswap.go --vy uniswap_exchange.vy        
Failed to build Vyper contract: vyper: exit status 1
/home/gballet/.local/lib/python3.6/site-packages/vyper/types/types.py:459: DeprecationWarning: Mapping definitions using subscript have deprecated (see VIP564). Use map(type1, type2) instead.
  DeprecationWarning
/usr/lib/python3/dist-packages/apport/report.py:13: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import fnmatch, glob, traceback, errno, sys, atexit, locale, imp
vyper.exceptions.InvalidTypeException: line 24:10 Unknown list type.
     23 totalSupply: public(uint256)                      # total number of UNI in existence
---> 24 balances: uint256[address]                        # UNI balance of an address
------------------^
     25 allowances: (uint256[address])[address]           # UNI allowance of one address on another
```

**Comment by gballet** [comment]
I was able to confirm and reproduce the issue with `1.9.8-unstable` and vyper `0.1.0b14+commit.49b201c`. Working on a fix.

**Comment by lmittmann** [comment]
for the sake of completeness: I used the `uniswap_exchange.vy` from the PR Uniswap/contracts-vyper/pull/44.

**Comment by lmittmann** [comment]
`abigen` also tries to include the files path in the variable and type names.

If you run
```
abigen --pkg main --type Uniswap --out uniswap.go --vy path/uniswap_exchange.vy
```
`aibgen` attempts to create variables like e.g., `Path/UniswapExchange.vyABI`.

**Comment by gballet** [comment]
Fixed in #20419
